src/fhost_fpga.py

```diff
@@ -150,36 +157,63 @@ def __init__(self, host, katcp_port=7147, boffile=None, connect=True, config=Non
             self.min_load_time = None
             self.network_latency_adjust = None
 
+        self._pfb_of_counts = {}
+        self._ct_counts = {}
+        self._qdr_counts = {}
+
     @classmethod
     def from_config_source(cls, hostname, katcp_port, config_source):
         boffile = config_source['bitstream']
         return cls(hostname, katcp_port=katcp_port, boffile=boffile,
                    connect=True, config=config_source)
 
-    def ct_okay(self, sleeptime=1):
+    def ct_okay(self, wait_time=1):
         """
-        Is the corner turner working?
-        :return: True or False,
+        Checks if the corner turner working.
+        :param: wait_time : Float or None : If not None, fetch ct counter,
+            wait this long, and fetch a second value; Else, use last read value
+            from cache. Value is not cached if wait_time is None.
+        :return: bool : True, if corner tuner is okay
```

Comment:
I guess the idea is to have no side effect if someone specifies a wait
time. Perhaps I'm overthinking things? Removing that logic would make the
code simpler too which is always a good thing :)

On Thu, Oct 1, 2015 at 2:48 PM, paulprozesky <notifications@github.com>
wrote:

> In src/fhost_fpga.py
> <https://github.com/ska-sa/corr2/pull/92#discussion_r40908739>:
>
> >          """
> > -        Is the corner turner working?
> > -        :return: True or False,
> > +        Checks if the corner turner working.
> > +        :param: wait_time : Float or None : If not None, fetch ct counter,
> > +            wait this long, and fetch a second value; Else, use last read value
> > +            from cache. Value is not cached if wait_time is None.
> > +        :return: bool : True, if corner tuner is okay
>
> In addition, why not cache when a wait_time is used? Just always cache the
> last read value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ska-sa/corr2/pull/92/files#r40908739>.
>